File: Assignments/Assignment4/Programming/BitAnd.java
import java.util.Scanner;

public class BitAnd {

    public static int bitAnd(int x, int y) {
        return ~(~x | ~y);
    }

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        System.out.print("Enter x: ");
        int x = sc.nextInt();

        System.out.print("Enter y: ");
        int y = sc.nextInt();

        int result = bitAnd(x, y);

        System.out.println("bitAnd(" + x + ", " + y + ") = " + result);

        sc.close();
    }
}

File: Assignments/Assignment4/Programming/BitAnd.c
#include <stdio.h>

int bitAnd(int x, int y) {
    return ~(~x | ~y);
}

int main() {

    int x, y;

    printf("Enter x: ");
    scanf("%d", &x);

    printf("Enter y: ");
    scanf("%d", &y);

    int result = bitAnd(x, y);

    printf("bitAnd(%d, %d) = %d\n", x, y, result);

    return 0;
}
